src/app/Upload.svelte

```diff
@@ -44,14 +54,41 @@
       for (let file of input.files) {
         files.set(file.name, file);
         saveToIndexedDB(file.name, file);
+        compileModel(file.name, file);
       }
     }
   };
 
   const saveToIndexedDB = (name: string, file: File) => {
     const worker = new Worker("worker.js");
-    worker.postMessage({ name, file });
-    worker.onmessage = () => {
+    const request: SaveFileRequestPayload = {
+      type: EventType.SaveFileRequest,
+      name,
+      file,
+    };
+    worker.postMessage(request);
+    worker.onmessage = (event) => {
+      const response = event.data as EventPayload;
+      if (response.errorMessage) {
+        console.error(response.errorMessage);
+      }
+      worker.terminate();
+    };
+  };
+
+  const compileModel = (name: string, file: File) => {
+    const worker = new Worker("worker.js");
+    const request: CompileModelRequestPayload = {
+      type: EventType.CompileModelRequest,
+      name,
+      file,
+    };
+    worker.postMessage(request);
+    worker.onmessage = (event) => {
+      const response = event.data as CompileModelResponsePayload;
```

Comment:
I love how this receives the compiled model's code, and then just... throws it out 😆